### PR TITLE
Fix NPE for newly initialized git repo

### LINKED ISSUE #66: NullPointerException when building with newly-initialized, uncommitted git repo
Having just initialized my sandbox with a git repository I wanted to make sure that my project build before committing. This failed until I completed the repo initialization with a commit.

Relevant stack trace:
```
[ERROR] Failed to execute goal com.spotify:docker-maven-plugin:0.2.5:build (docker-build) on project docker-tomcat: Exception caught: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.spotify:docker-maven-plugin:0.2.5:build (docker-build) on project docker-tomcat: Exception caught
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:216)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:355)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:216)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:160)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Exception caught
	at com.spotify.docker.AbstractDockerMojo.execute(AbstractDockerMojo.java:65)
	at com.spotify.docker.BuildMojo.execute(BuildMojo.java:79)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 19 more
Caused by: java.lang.NullPointerException
	at com.spotify.docker.Git.getCommitId(Git.java:71)
	at com.spotify.docker.BuildMojo.execute(BuildMojo.java:209)
	at com.spotify.docker.AbstractDockerMojo.execute(AbstractDockerMojo.java:63)
	... 22 more
```

### PR DESCRIPTION
**Pull request by davidxia**
Fixes #66

### CODE REVIEW

File: src/main/java/com/spotify/docker/DockerBuildInformation.java [hunk 1]
```diff
@@ -64,7 +66,10 @@ private void updateGitInformation(Log log) {
       Repository repo = new Git().getRepo();
       if (repo != null) {
         this.repo   = repo.getConfig().getString("remote", "origin", "url");
-        this.commit = repo.resolve("HEAD").getName();
+        final ObjectId head = repo.resolve("HEAD");
+        if (head != null && isNullOrEmpty(head.getName())) {
```
Comment (gimaker) [inline src/main/java/com/spotify/docker/DockerBuildInformation.java]: This condition is true only if head.getName() is null or empty. I suspect there should be a logical not in front of the isNullOrEmpty() :)

Comment (davidxia) [inline src/main/java/com/spotify/docker/DockerBuildInformation.java]: good catch :)

**Comment by davidxia** [comment]
@gimaker

**Comment by gimaker** [comment]
Note a huge fan of making the class have mutable state just to enabling testing -- this suggests this needs refactoring to be more testable. However, it also doesn't look terribly bad so :+1:

**Pull request merged**